File: Algorithms - Design and Analysis/Week 6/two_sum.py
import sys

target_numbers = range(-10000,10001) # [-10000,10000] (inclusive)
numbers_to_use = {}
valid_targer_number_count = 0

for line in sys.stdin:
    numbers_to_use[int(line)] = True

for target_number in target_numbers:
    for number in numbers_to_use:
        if target_number - number != number and target_number - number in numbers_to_use :
            valid_targer_number_count+=1
            print("Find targer number {}".format(target_number))
            break

print(valid_targer_number_count)

File: Algorithms - Design and Analysis/Week 6/two_sum.cpp
#include <iostream>
#include <unordered_map>
using namespace std;
int main() {
  unordered_map<long long, bool> numbers_to_use;
  long long number;
  int target_count = 0;
  while (scanf("%lld", &number) != EOF) {
    numbers_to_use[number] = true;
  }
  for (int target = -10000; target <= 10000; target++) {
    for (auto& kv : numbers_to_use) {
      long long number_to_use = kv.first;
      if (target - number_to_use != number_to_use &&
          numbers_to_use.find(target - number_to_use) != numbers_to_use.end()) {
        target_count++;
        cout << "Find target number " << target << endl;
        break;
      }
    }
  }
  cout << "Total " << target_count << " valid targer numbers";
  // Answer 427
}
